bugsnag-android-core/src/main/java/com/bugsnag/android/AppDataCollector.kt

```diff
@@ -49,12 +49,32 @@ internal class AppDataCollector(
         )
     }
 
+    private fun getProcessImportance(): String? {
+        try {
+            return when (activityManager?.runningAppProcesses?.last()?.importance) {
+                ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND -> "foreground"
+                ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND_SERVICE -> "foreground service"
+                ActivityManager.RunningAppProcessInfo.IMPORTANCE_TOP_SLEEPING -> "top sleeping"
+                ActivityManager.RunningAppProcessInfo.IMPORTANCE_VISIBLE -> "visible"
+                ActivityManager.RunningAppProcessInfo.IMPORTANCE_PERCEPTIBLE -> "perceptible"
+                ActivityManager.RunningAppProcessInfo.IMPORTANCE_CANT_SAVE_STATE -> "can't save state"
+                ActivityManager.RunningAppProcessInfo.IMPORTANCE_SERVICE -> "service"
+                ActivityManager.RunningAppProcessInfo.IMPORTANCE_CACHED -> "cached"
+                ActivityManager.RunningAppProcessInfo.IMPORTANCE_GONE -> "gone"
+                else -> "unknown importance (${activityManager?.runningAppProcesses?.last()?.importance})"
```

Comment:
This should rather use `getMyMemoryState` to ensure we are getting the correct process info.

```suggestion
            val appInfo = ActivityManager.RunningAppProcessInfo()
            ActivityManager.getMyMemoryState(appInfo)
            
            if (appInfo.pid == 0) {
                return null
            }
            
            return when (appInfo.importance) {
                ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND -> "foreground"
                ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND_SERVICE -> "foreground service"
                ActivityManager.RunningAppProcessInfo.IMPORTANCE_TOP_SLEEPING -> "top sleeping"
                ActivityManager.RunningAppProcessInfo.IMPORTANCE_VISIBLE -> "visible"
                ActivityManager.RunningAppProcessInfo.IMPORTANCE_PERCEPTIBLE -> "perceptible"
                ActivityManager.RunningAppProcessInfo.IMPORTANCE_CANT_SAVE_STATE -> "can't save state"
                ActivityManager.RunningAppProcessInfo.IMPORTANCE_SERVICE -> "service"
                ActivityManager.RunningAppProcessInfo.IMPORTANCE_CACHED -> "cached"
                ActivityManager.RunningAppProcessInfo.IMPORTANCE_GONE -> "gone"
                else -> "unknown importance (${appInfo.importance})"
```